pandas/core/groupby/__init__.py

```diff
@@ -1,15 +1,12 @@
-from pandas.core.groupby.generic import (
-    DataFrameGroupBy,
-    NamedAgg,
-    SeriesGroupBy,
-)
-from pandas.core.groupby.groupby import GroupBy
-from pandas.core.groupby.grouper import Grouper
-
-__all__ = [
-    "DataFrameGroupBy",
-    "NamedAgg",
-    "SeriesGroupBy",
-    "GroupBy",
-    "Grouper",
-]
+from __future__ import annotations
+
+from typing import Any
+
+from pandas._core import groupby as groupby_
+from pandas.core.common import _depr_core
```

Comment:
In the future, will this raise it's own `DeprecationWarning` if this is imported from `pandas.core`?

Reply:
Once we deprecate `pandas.core.common`, I think we'd move this method to `pandas._core.common`.